.github/workflows/sbom.yaml

```diff
@@ -0,0 +1,78 @@
+name: Generate SBOM
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
It would be good to check with @cmur2 whether this should be part of the [unified ci](https://github.com/camunda/camunda/wiki/CI-&-Automation#unified-ci).

Reply:
@korthout I would argue that is kind of wasteful to run this on every PR and maybe even on every merge into `main`

The purpose of this pipeline is just extract our 3rd party deps for compliance reasons and we would only need it a couple of times per year, if this breaks we would have plenty of time to fix this.

In my opinion this workflow should have a manual trigger

Reply:
>In my opinion this workflow should have a manual trigger

In that case, it must at least be tested regularly to ensure it doesn't block a release, e.g. by the release dry-run. It would be good if @cmur2 can comment on this.

Reply:
>Alternatively, it can be run on: schedule every week or so.

If it fails, who'd be paged to look into it?

Reply:
> Is it needed on every minor release? And/or on every alpha release?

For now only on minors when we update the docs

> If it fails, who'd be paged to look into it?

This is more of a company-wide thing but I suppose it's ok to ping @nathansandi and me for now

@korthout @cmur2 I updated the triggers to add a manual and a schedule dispatch. You can check my last commit: Is it needed on every minor release? And/or on every alpha release?